app/src/main/java/com/eveningoutpost/dexdrip/ui/helpers/BitmapUtil.java

```diff
@@ -86,6 +56,84 @@ public static Bitmap getTiled(final Bitmap source, final int xSize, final int yS
         return tiledBitmap;
     }
 
+    public static Bitmap getBackgroundBitmap(final int xSize, final int ySize, final String background_file) {
+        Bitmap tiledBitmap = null;
+        try {
+            tiledBitmap = BitmapLoader.bitmapFromBundleCache(background_file);
+            if (tiledBitmap == null) {
+                android.util.Log.d("NumberWall", "Regenerating image");
+
+                final Uri background_uri = Uri.parse(background_file);
+
+                Bitmap image_bitmap = null;
+                if (background_file.startsWith("content://")) {
+                    FileDescriptor fileDescriptor = xdrip.getAppContext().getContentResolver().openFileDescriptor(background_uri, "r").getFileDescriptor();
+                    image_bitmap = BitmapFactory.decodeFileDescriptor(fileDescriptor);
+                } else {
+                    File file = new File(background_file);
+                    FileInputStream in = new FileInputStream(file);
+                    image_bitmap = BitmapFactory.decodeStream(in);
+                    in.close();
+                }
+
+                Bitmap rotated_bitmap = image_bitmap;
+                final Matrix image_matrix = new Matrix();
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
+                    ExifInterface exif = null;
+                    if (background_file.startsWith("content://")) {
+                        FileDescriptor fileDescriptor = xdrip.getAppContext().getContentResolver().openFileDescriptor(background_uri, "r").getFileDescriptor(); // reset
+                        exif = new ExifInterface(fileDescriptor);
+                    } else {
+                        File file = new File(background_file);
+                        FileInputStream in = new FileInputStream(file);
+                        exif = new ExifInterface(in);
+                        in.close();
```

Comment:
If you close the input stream here, you cannot access the exif data anymore. Please change it and use proper exception handling (try, catch finally). An example can be found at https://blog.mindorks.com/exifinterface